src/Castle.Windsor/MicroKernel/Registration/Types.cs

```diff
@@ -108,12 +108,20 @@ public static FromAssemblyDescriptor FromAssemblyInDirectory(AssemblyFilter filt
 		///     <c>MyApp.SuperFeatures.dll</c>, <c>mscorlib.dll</c> and <c>ThirdPartyCompany.UberControls.dll</c> the <c>MyApp.exe</c> and <c>MyApp.SuperFeatures.dll</c> will be scanned for components, and other
 		///     assemblies will be ignored.
 		/// </remarks>
+#if FEATURE_GETCALLINGASSEMBLY
 		[MethodImpl(MethodImplOptions.NoInlining)]
 		public static FromAssemblyDescriptor FromAssemblyInThisApplication()
 		{
 			var assemblies = new HashSet<Assembly>(ReflectionUtil.GetApplicationAssemblies(Assembly.GetCallingAssembly()));
 			return new FromAssemblyDescriptor(assemblies, null);
```

Comment:
Could we change the implementation of this method to just call the new one below and pass `GetCallingAssembly()`.